wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -160,46 +161,40 @@ protected void setBuilder(Builder builder) {
     return adapter.fromInt(value);
   }
 
+  /**
+   * Create a {@code byte[]} of the serialized form of the data.
+   * <p>
+   * Note: This is an inefficient means of serializing data. For best performance, call
+   * {@link Wire#adapter} and then {@link MessageAdapter2#write}.
+   */
   @SuppressWarnings("unchecked")
   public byte[] toByteArray() {
```

Comment:
It's simple but it's the wrong way into the API. This method is doubly inefficient. It requires a parallel `Wire` instance, its backing maps, and its own message adapter instances which are massive objects to keep around. This is a waste of memory when we can re-use the main instance which is also more knowledgable about the extensions. This method also buffers the output into a `byte[]` when you are almost always writing to something better observed as a stream.

As to the base class, it's a convenience for simplifying the generated code but I don't see how that requires this method be here–especially since it has no effect on the API for deserialization.